R/nearest_stations_imgw.R

```diff
@@ -60,7 +60,7 @@ nearest_stations_imgw = function(type = "meteo",
     stop("You've provided wrong type argument; please use: \"meteo\", or \"hydro\"")
   }
 
-  if (dim(result)[1] == 0) {
+  if (nrow(result) == 0) {
```

Comment:
what's the rationale behind this change?

Reply:
`result` is a simple data.frame so in my opinion it is better to make it clear (for readability) that we are using the number of rows. `dim()` can refer to multidimensional arrays but behind the code `nrow()` is just `dim()[1]`.

Reply:
I think both approaches are fine. I don't remember precisely, but maybe we wanted to have a passing code for arrays that might be converted to vector and `dim` would work for any class of object, but I am only guessing...

Reply:
OK, let's stay with `dim(result)[1]` to avoid changing too much. BTW: Using `dim()` with objects of another class (not matrices / data frames) can produce results that are completely unexpected by us. For example, stars object from [stars](https://r-spatial.github.io/stars/) package. It uses arrays on backend but first dimension are attributes, second are columns, third are rows, four are bands, etc.